superset/db_engine_specs/hana.py

```diff
@@ -42,9 +42,24 @@ class HanaEngineSpec(PostgresBaseEngineSpec):
 
     @classmethod
     def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
+        """
+        Tested: hana time returns DATETIME and STRING
+        HANA:
+        :return dttm->DATETIME
+        HANA->DATE:f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
+
+        :return dttm->DATETIME
+        HANA->TIMESTAMP:
+        f"TO_TIMESTAMP('{dttm.isoformat(timespec="microseconds")}',
+        'YYYY-MM-DD"T"HH24:MI:SS.ff6')"
+
+        :return dttm->STRING
+        HANA->NVARCHAR TYPE:f"TO_CHAR('{dttm.date().isoformat()}', 'YYYYMMDD')"
+        """
         tt = target_type.upper()
-        if tt == "DATE":
-            return f"TO_DATE('{dttm.date().isoformat()}', 'YYYY-MM-DD')"
-        if tt == "TIMESTAMP":
-            return f"""TO_TIMESTAMP('{dttm.isoformat(timespec="microseconds")}', 'YYYY-MM-DD"T"HH24:MI:SS.ff6')"""  # pylint: disable=line-too-long
+        if tt == "DATETIME":
```

Comment:
This doc lists `DATE, TIME, SECONDDATE, TIMESTAMP` as the date-related types: https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.03/en-US/20a1569875191014b507cf392724b7eb.html

I'm not sure how the dialect/dbapi driver works, maybe the types get translated somewhere.

It seems like there should be a catchall `else:` at the end, even if it's raising a `ValueError` or `NotImplementedError` to say something like "target data type {type} isn't handled by the DBEngineSpec for SAP Hana" or something like that...

Reply:
![image](https://user-images.githubusercontent.com/44701753/69016920-f4573600-09dd-11ea-97f0-04c481efbd4c.png)
![image](https://user-images.githubusercontent.com/44701753/69016787-d1785200-09dc-11ea-9da9-0dfe7e34d7f3.png)

HANA does have these types, time function returns is indeed a DATETIME type (except the time type, time is short format), so the judge tt = = DATETIME can satisfy time format, tt = = data STRING because engineers don't have the default time default store time as a STRING (SAP) is converted into a time format in order to support the filtration of time, before I have already explained.The above has worked well in my formal environment after several rounds of testing